core/controllers/acl_decorators.py

```diff
@@ -2263,12 +2282,17 @@ def test_can_change_topic_publication_status(self, **kwargs):
         if not self.user_id:
             raise self.NotLoggedInException
 
+        try:
+            topic_domain.Topic.require_valid_topic_id(topic_id)
```

Comment:
Instead of this check, I was expecting to have a check which not just validates the pattern of the topic ID it should (also) validate that a topic exist with the given ID. (see line 2306)

@aks681  if this acl is just about the user's general rights (not specific to the given topic) then should we remove check related to a specific topic over here! (not sure!)
![image](https://user-images.githubusercontent.com/16653571/76241378-df3b3700-625a-11ea-8998-3175c9dbd089.png)

Reply:
>Instead of this check, I was expecting to have a check which not just validates the pattern of the topic ID it should (also) validate that a topic exist with the given ID. (see line 2306)

Looks like we check if the topic exists in the PUT method in this case.

> if this acl is just about the user's general rights (not specific to the given topic) then should we remove check related to a specific topic over here! (not sure!)

It seems to simply check if the user has the "ACTION_CHANGE_TOPIC_STATUS" role. I think in this case we could move this validation into the put method. What do you think @DubeySandeep?

Reply:
The function `require_valid_topic_id` is in the domain layer, and hence only the checks present in it currently is possible, the other checks, as @kevintab95 mentioned, is present in the controller layer.
In the screenshot, isn't the existence of exploration checked when the rights model is fetched?

> if this acl is just about the user's general rights (not specific to the given topic) then should we remove check related to a specific topic over here! (not sure!

Yeah, moving that out of here seems fine, since we need to fetch the topic_rights object in the controller anyway.